test/string/test_string_to_number.f90

```diff
@@ -0,0 +1,343 @@
+        if (allocated(error)) return
+
+        call check(error, ucheck("0.1234E0"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("12.34E0"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("0.34E2"))
+        if (allocated(error)) return
+
+        call check(error, ucheck(".34e0"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("34.E1"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("-34.5E1"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("0.0021E10"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("12.21e-1"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("12.21e+001 "))
+        if (allocated(error)) return
+
+        call check(error, ucheck("-1"))
+        if (allocated(error)) return
+
+        call check(error, ucheck(" -0.23317260678539647E-01 "))
+        if (allocated(error)) return
+
+        call check(error, ucheck(" 2.5647869e-003 "//char(13)//char(10)))
+        if (allocated(error)) return
+
+        call check(error, ucheck("1.-3"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("Inf"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("-Inf"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("NaN"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("0.123456789123456789123456789123456789"))
+        if (allocated(error)) return
+
+        call check(error, ucheck("1234567890123456789012345678901234567890-9") )
+        if (allocated(error)) return
+
+        call check(error, ucheck("123456.78901234567890123456789012345678901234567890+2") )
+        if (allocated(error)) return
+
+        call check(error, ucheck("0.140129846432481707092372958328991613128026194187651577"//&
+        &                        "175706828388979108268586060148663818836212158203125E-44"))
+        if (allocated(error)) return
+
+    contains
+        logical function ucheck(s)
+            character(*), intent(in) :: s
+            real(wp) :: formatted_read_out
+            real(wp) :: to_num_out
+            real(wp) :: abs_err
+            real(wp) :: rel_err
+
+            ucheck = .true.
+            read(s,*) formatted_read_out
+            to_num_out = to_num(s, to_num_out)
+            abs_err = to_num_out - formatted_read_out
+            rel_err = abs_err / formatted_read_out
+
+            if(abs(rel_err) > epsilon(0.0_wp)) then
+                write(*,"('formatted read : ' g0)") formatted_read_out
+                write(*,"('to_num         : ' g0)") to_num_out
+                write(*,"('difference abs : ' g0)") abs_err
+                write(*,"('difference rel : ' g0 '%')") rel_err * 100
+                ucheck = .false.
+            end if
+        end function
+    end subroutine
+
+    subroutine test_to_quadruple(error)
```

Comment:
It seems to be 3 copy-paste. If so, then a `fypp` loop could be implemented.

Reply:
Done